packages/analytics-js-integrations/src/integrations/Criteo/utils.js

```diff
@@ -185,6 +185,33 @@ const processViewedCartEvent = (finalPayload, productInfo) => {
   finalPayload.push(viewBasketObject);
 };
 
+/**
+ * Adds a product to the cart and updates the final payload.
+ * 
+ * @param {Object} properties - The properties of the event.
+ * @param {Array} finalPayload - The final payload array to be updated.
+ * @param {Array} productInfo - The information of the product to be added to the cart.
+ * @returns {void}
+ */
+const handleProductAdded = (message,finalPayload) => {
+  const buildProductObject = (properties) => 
+    [
+      {
+        id: String(properties.product_id),
+        price: parseFloat(properties.price),
+        quantity: parseInt(properties.quantity, 10),
+      }
+    ]
+  
+  const {properties} = message;
+  const addToCartObject = {
+    event: 'addToCart',
+    currency: properties.currency,
+    item: validateProduct(properties, 0) ? buildProductObject(properties) : [] ,
+  };
```

Comment:
We need to iterate over each product and run validateProduct isn't it? We are doing similar things for other events as well.

Reply:
we are mapping product added to add to cart, and it has no product array .ref : https://www.rudderstack.com/docs/event-spec/ecommerce-events-spec/ordering/#product-added